float32 distance
float32 heading
int32 cov_size
float32[] covariance